test/ecto/adapters/sqlite3_test.exs

```diff
@@ -51,4 +59,27 @@ defmodule Ecto.Adapters.SQLite3Test do
       File.rm(opts[:database])
     end
   end
+
+  describe ".autogenerate/1" do
+    test ":id must be generated from storage" do
+      assert SQLite3.autogenerate(:id) == nil
+    end
+
+    test ":embed_id is a UUID in string form" do
+      assert string_uuid?(SQLite3.autogenerate(:embed_id))
+    end
+
+    test ":binary_id with type :string is a UUID in string form" do
+      Application.put_env(:ecto_sqlite3, :binary_id_type, :string)
+      assert string_uuid?(SQLite3.autogenerate(:binary_id))
+    end
+
+    test ":binary_id with type :binary is a UUID in binary form" do
+      Application.put_env(:ecto_sqlite3, :binary_id_type, :binary)
+      assert binary_uuid?(SQLite3.autogenerate(:binary_id))
+    end
+  end
+
+  defp string_uuid?(uuid), do: Regex.match?(@uuid_regex, uuid)
+  defp binary_uuid?(uuid), do: bit_size(uuid) == 128
```

Comment:
Actually validating whether the values are proper UUIDs or not would obviously be quite a bit more involved than this but I figured that those simple checks should be enough for the purposes of these tests.

Reply:
This is okay.